firestore/cloud-client/snippets.py

```diff
@@ -29,15 +29,15 @@ def quickstart_new_instance():
     return db
 
 
-def quickstart_add_data_one():
+def quickstart_add_data_one(collection_id, document_id, data):
     db = firestore.Client()
     # [START quickstart_add_data_one]
-    doc_ref = db.collection(u'users').document(u'alovelace')
-    doc_ref.set({
-        u'first': u'Ada',
-        u'last': u'Lovelace',
-        u'born': 1815
-    })
+    # collection_id = "users"
```

Comment:
The problem with this approach is that these comments are fundamentally not tested.

Here is an idea. Can you have a docstring with these value assignment examples, then in the test, extract those value assignments and test with those values?

Reply:
I've tried to extract docstrings with `inspect` and run them with `eval/exec`. It doesn't look like a good variant, as because of scopes there is no way to write something universal - to parse all the snippets docstrings with one helper. And it actually looks ugly. So, for now I don't see how we can do it.